core/modules/keyboard.js

```diff
@@ -322,8 +322,21 @@ KeyboardManager.prototype.updateShortcutLists = function(tiddlerList) {
 };
 
 KeyboardManager.prototype.handleKeydownEvent = function(event) {
+	return this.handleKeydownEventInternal(event, false);
+}
+
+KeyboardManager.prototype.handlePrimaryKeydownEvent = function(event) {
+	return this.handleKeydownEventInternal(event, true);
+}
+
+KeyboardManager.prototype.handleKeydownEventInternal = function(event, onlyPrimary) {
 	var key, action;
 	for(var i=0; i<this.shortcutTiddlers.length; i++) {
+		var tiddler = $tw.wiki.getTiddler(this.shortcutTiddlers[i]);
+		if (onlyPrimary && !tiddler.fields.primary) {
```

Comment:
I think this test is not specific enough. It only tests for the existence of the primary field. 
IMO it should be `if (onlyPrimary && ( tiddler.fields.primary === "yes")` ... See `yes` instead of `on`. 
I'm pretty sure most of the times we use "yes" and "no" for options like this one.